File: src/ders8/rpgornekjava/Buyucu.java
package ders8.rpgornekjava;

public class Buyucu extends Oyuncu implements BuyuYapabilir {

    public Buyucu(String isim, int saglik) {
        super(isim, saglik);
    }

    public Buyucu(String isim) {
        super(isim);
        this.saglik = 20;
    }

    @Override
    public void buyuYap() {
        System.out.println("Büyücü büyü yapıyor...");
    }
}

File: src/ders8/rpgornekkotlin/Buyucu.kt
package ders8.rpgornekkotlin

class Buyucu(isim: String, saglik: Int = 50) : Oyuncu(isim, saglik), BuyuKullanabilir{
    override fun buyuYap() {
        println("$isim büyü yapıyor...")
    }

}